### ISSUE TITLE
Convert proxied Arrays

### ISSUE THREAD

**Issue by tomatenbaumful** [altmp/altv-js-module-v2#111]
When converting JS Proxys to MValues currently it gets handled by the else in the convert function.
So an normal Object is converted to an Dict wich is expected, but an Array will be converted to {} since the keys cant be read in v2 (in v1 the Array would be converted to an Object with  the indece as keys).
Having the array be handled would be more intuitive.

Client:
```js
/// <reference path="./index.d.ts"/>
import * as alt from "@altv/client"

const array = ["Hello", "World"];
const proxy = new Proxy(array, {
  get: (target, prop) => {
    alt.log(`Getting property "${prop}"`);
    return target[prop];
  }
});
const object = {
  hello: "world"
};
const proxy2 = new Proxy(object, {
  get: (target, prop) => {
    alt.log(`Getting property "${prop}"`);
    return target[prop];
  }
});
alt.Events.emitServer("test", proxy, proxy2);
```
Server:
```js
/// <reference path="./index.d.ts"/>
import * as alt from "@altv/server";

alt.Events.onRemote("test", (player, args) => {
  alt.log(args);
});
```
 Output:

>[ {}, [ 'Hello', 'World' ] ]

Could be fixed with an aditional check for an proxied Array:
```c++
else if (val->IsProxy() && val.As<v8::Proxy>()->GetTarget()->IsArray())
        {

                Logger::Error("Proxied Array");
                auto arr = val.As<v8::Array>();
                uint32_t length = val.As<v8::Proxy>()->Get(ctx, JSValue("length")).ToLocalChecked()->Uint32Value(ctx).ToChecked();
                Logger::Error("Length: " , length);
                alt::MValueList list = core.CreateMValueList(length);
                for(uint32_t i = 0; i < length; ++i)
                {
                    v8::Local<v8::Value> value;
                    if(!arr->Get(ctx, i).ToLocal(&value)) continue;
                    list->Set(i, JSToMValue(value, allowFunction));
                }

                return list;


        }
```

**Comment by LeonMrBonnie** [comment]
Just don't pass proxies to events/meta. It's not made to store any complex data types and passing a data type that isn't supported is the fault of your own code.

**Comment by Yiin** [comment]
@LeonMrBonnie if it can be fixed as the proposed code suggests, argument that "it's not made to store complex data types" doesn't really hold, the API is just bad in that case.

**Comment by tomatenbaumful** [comment]
The issue i have does not necessarily mean i want to send them or that i do, but it was an issue that is hard to debug since the conversion just defaults to something and does not tell you the data type is unsupported. 
It just would be a better experience for developers to get told hey you cant send this then to just get get incorrectly serialized or missing data. 
This isnt only an issue for proxied arrays but with all non serializable data!

**Comment by LeonMrBonnie** [comment]
Fixed in c46a50b7d6e79e7a368e28030df1b6a49db8940c
